subsys/bluetooth/host/hci_core.c

```diff
@@ -2621,21 +2623,25 @@ static void read_buffer_size_v2_complete(struct net_buf *buf)
 	LOG_DBG("status %u", rp->status);
 
 #if defined(CONFIG_BT_CONN)
-	bt_dev.le.acl_mtu = sys_le16_to_cpu(rp->acl_max_len);
-	if (!bt_dev.le.acl_mtu) {
+	uint16_t acl_mtu = sys_le16_to_cpu(rp->acl_max_len);
+
+	if (!acl_mtu || !rp->acl_max_num) {
 		return;
```

Comment:
Do you know why we are returning here?

In the case that we have a host with ACL and ISO support, and a controller with just broadcast ISO support, we won't have any ACL buffers from the controller, but still ISO buffers, but since we are returning here, the ISO buffers would never be initialized.

Should we instead of  
```c
if (!acl_mtu || !rp->acl_max_num) {
	return;
}
bt_dev.le.acl_mtu = acl_mtu;
k_sem_init(&bt_dev.le.acl_pkts, rp->acl_max_num, rp->acl_max_num);
```

have 
```c
if (acl_mtu && rp->acl_max_num) {
	bt_dev.le.acl_mtu = acl_mtu;
	k_sem_init(&bt_dev.le.acl_pkts, rp->acl_max_num, rp->acl_max_num);
}
```

?

Ditto for ISO in case we ever want to add anything additional after ISO.

Reply:
@jhedberg Do you know if there is a good reason to return? The proposal seems fair.

Reply:
@theob-pro yes, the proposal from @Thalley makes sense. I hadn't previously looked in detail of the code flow for the new ISO-related v2 command.